php/installer/Database.class.inc

```diff
@@ -45,6 +45,163 @@ class Database
     }
 
     /**
+     * On executing `SHOW GRANT;`, rows with a single column are returned.
+     * Each row's value is a string that lists the MySQL user's privileges for
+     * a given table.
+     *
+     * @param string $raw The string of each row returned by `SHOW GRANT;`
+     *
+     * @see https://dev.mysql.com/doc/refman/5.7/en/show-grants.html
+     *
+     * @return object The parsed privilege for a string returned by `SHOW GRANT;`
+     */
+    private function _parsePrivilege($raw)
+    {
+        $token_arr = explode(" ON ", $raw);
+        if (count($token_arr) != 2) {
+            //Avoid letting the user see the "privileges", output to error_log()
+            error_log(
+                "Database::_parsePrivilege():Received {$raw}, could not split in two"
+            );
+            throw new Exception("Could not parse privilege");
+        }
+        $database_table_end = strpos($token_arr[1], " ");
+        $database_table     = substr($token_arr[1], 0, $database_table_end);
+
+        $privilege_arr_raw = substr($token_arr[0], strlen("GRANT "));
+        $privilege_arr     = explode(", ", $privilege_arr_raw);
+
+        return (object)array(
+                        "database_table" => $database_table,
+                        "privilege_arr"  => $privilege_arr,
+                       );
+    }
+    private $_privileges = null;
```

Comment:
It is only ever used by one method, the one directly below it. It should not be used anywhere else and isn't. Putting it just above the method makes it clearer that it's only relevant around the area, rather than applicable to the whole class.